text/0096-contract-metadata.md

```diff
@@ -0,0 +1,213 @@
+
+## Recommended Client Usage (Wallets, Explorers, Indexers)
+
+[recommended-client-usage-wallets-explorers-indexers]: #recommended-client-usage-wallets-explorers-indexers
+
+To adopt this standard, tools and services in the Casper ecosystem should:
+
+- Detect compliance by checking for named keys (`name`, `description`, `icon_uri`, `project_uri`) in the contract's state via Casper RPC (`query_global_state`).
+- Directly query metadata via the Casper RPC interface without performing contract calls, leveraging the efficient, immutable state queries.
+- Display available metadata fields clearly:
+  - Use `name` and `icon_uri` prominently (e.g., wallet contract lists, explorer pages).
+  - Use `description` and `project_uri` in detailed views for contract transparency and user education.
+- Handle missions fields, since all fields are optional, gracefully handle `None` values.
+- Cache metadata permanently after initial retrieval since immutability is guaranteed.
+
+## Encouraging Community Adoption
+
+[encouraging-community-adoption]: #encouraging-community-adoption
+
+To maximize impact, we recommend:
+
+- Official Casper documentation encourage developers to follow this metadata standard. 
+- Frameworks and templates (like Odra) adopt this pattern by default.
+- Wallets, explorers, and indexing services integrate support, making metadata discoverable and meaningful for end-users.
+
+## Conclusion
+
+[conclusion]: #conclusion
+
+This Casper On-Chain Contract Metadata Standard, using named URef keys, provides a simple yet powerful mechanism for contracts to publish immutable metadata directly on-chain. By standardizing metadata storage and access, the Casper community enhances transparency, usability, and integrity throughout the ecosystem. Broad adoption by contract developers and integration by ecosystem tools will significantly improve discoverability and user trust, positioning Casper at the forefront of user-friendly blockchain ecosystems.
+
+## Appendix
+
+[appendix]: #appendix
+
+### Changelog
+
+[changelog]: #changelog
+
+- v1.0: Initial draft including CEP96ContractMetadata interface, and backend indexing guidelines.
+
+### References
+
+[references]: #references
+
+- [Casper Documentation](https://docs.casper.network/)
+- [CEP-78 (Enhanced NFT Standard)](https://github.com/casper-ecosystem/cep-78-enhanced-nft)
+- [ERC-721 (Ethereum NFT Standard)](https://eips.ethereum.org/EIPS/eip-721)
+- [Casper RPC Methods](https://docs.casper.network/developers/json-rpc/minimal-compliance)
+- [Casper Smart Contracts Guide](https://docs.casper.network/concepts/smart-contracts)
+- [CasperLabs Main GitHub](https://github.com/CasperLabs)
```

Comment:
Those references seem to be irrelevant. I'd reference only the CEP95 NFT here since it's the only place where this standard is mentioned